.circleci/config.yml

```diff
@@ -253,6 +253,104 @@ jobs:
                 include_job_number_field: false
                 include_project_field: false
                 failure_message: ':red_circle: Scheduled tests failed on iOS device!'
+  android-build-and-publish-react-native-bridge:
+    # This job builds the JS bunle required by the Android version of the
+    # react-native-bridge package, builds the AAR for that package, and uploads
+    # it to Bintray.
+    #
+    # Running on a Ubuntu machine to have more memory compared to the Docker +
+    # Node or Docker + Android images that CircleCI offers (circleci/node)
+    #
+    # Also running all in one job so that we can bail out early under cetrain
+    # conditions and save CI cycles.
+    #
+    # See: https://github.com/wordpress-mobile/gutenberg-mobile/pull/3038#issuecomment-765334675
+    machine:
+      image: << pipeline.parameters.linux-machine-image >>
+    steps:
+      - run:
+          name: Evaluate whether to run
+          command: |
+            # This is a partial duplication of the logic to compute the version
+            # to use for the AAR from below. It's useful to have it here
+            # because there are conditions under which we shouldn't run the job
+            # that can't be captured in a CircleCI filter.
+            #
+            # Exiting here saves a non-trivial amount of CI time because of how
+            # heavy the repo is to checkout (it takes at least 1 minute) and
+            # how time consuming running `npm ci` is.
+            if [[ -z ${CIRCLE_TAG+x} ]]; then
+              if [[ -n "$CIRCLE_PULL_REQUEST" ]]; then
+                echo "Build initiated from a branch with a pull request: proceeding..."
+              elif [[ $CIRCLE_BRANCH == "develop" ]]; then
+                echo "Build initiated on $CIRCLE_BRANCH: proceeding..."
+              else
+                # We end up here on the first push of a new branch, when there
+                # isn't a PR open for it yet.
+                echo "Build initiated from a feature branch with no pull request: aborting."
+                circleci-agent step halt
```

Comment:
[This build](https://app.circleci.com/pipelines/github/wordpress-mobile/gutenberg-mobile/10976/workflows/b36695d1-4e5b-4fe1-9333-da8769a99a7d/jobs/57726) is an example of this scenario

![image](https://user-images.githubusercontent.com/1218433/111438120-fd821f80-8757-11eb-8cb6-d4cfbfbfc50d.png)

Unfortunately, CircleCI doesn't treat halts differently and the UI might be misleading as it looks like the build passed.